### DISCUSSION TITLE
Dividing a data object into batches

### DISCUSSION THREAD

**Discussion by atul1234anand** [pyg-team/pytorch_geometric#9207] [Q&A]
Is there a way to divide a data object in pytorch geometric into batches without sampling?

**Comment by rusty1s** [comment]
Do you mean into a set of subgraphs? Take a look at `NeighborLoader`.

**Reply by atul1234anand** [reply]
NeighbourLoader samples some nodes per neighbourhood right? I dont want that. I would like to sample a bunch of nodes and the induced graph of those nodes.

**Reply by rusty1s** [reply]
If you don't want to sample a neighborhood, you can also do
```python
nodes = torch.randint(0, data.num_nodes, (batch_size, ))
batch = data.subgraph(nodes)
```
